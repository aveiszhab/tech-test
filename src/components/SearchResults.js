import React from 'react';
import '../styles/SearchResults.css';

const SearchResults = ({results}) => {
    if(!results.length) {
       return <h1 className='no-result'>No result</h1>
    } else {
    return(
        <div className='image-container'>
            <>
                {results.map(image => 
                    <img 
                    key={image}
                    src={image}
                    alt='spaceImage'
                    className='spaceImages'
                    />
                )}
            </>            
        </div>
    )
    };
};
    
    

export default SearchResults;