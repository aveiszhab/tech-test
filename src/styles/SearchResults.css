.image-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
}

.spaceImages {
    max-height: 15vw;
    min-width: 20%;
    object-fit: cover;
    vertical-align: bottom;
    margin: 2%;
    border-radius: 8px;
}

.no-result {
    align-self: center;
    font-size: 2vw;
    font-weight: 500;
}