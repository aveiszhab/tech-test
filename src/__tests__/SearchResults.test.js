import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchResults from '../components/SearchResults';


describe('SearchResults', () => {
    it('renders correctly to match the SearchResults snapshot', () => {
        const results = ['img1', 'img2']
        const {asFragment} = render(<SearchResults results={results}/>);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders SearchResults correctly if there is a result from API', () => {
        const results = ['img1','img2']
        
        render(<SearchResults results = {results}/>);
            const images = screen.getAllByRole('img');
        
            expect(images).toHaveLength(2);      
    });
    
      it('renders Searchresults when the result is an empty array from API', () => {
            const results = [];
        
            render(<SearchResults results = {results}/>);
        
            const message = screen.getByRole('heading');
            const image = screen.queryByRole('img');
        
            expect(image).not.toBeInTheDocument();
            expect(message).toHaveClass('no-result');
            expect(message).toHaveTextContent(/no result/i);
      });
});