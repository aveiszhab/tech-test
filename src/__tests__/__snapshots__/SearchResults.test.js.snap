// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly to match the SearchResults snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="spaceImage"
      class="spaceImages"
      src="img1"
    />
    <img
      alt="spaceImage"
      class="spaceImages"
      src="img2"
    />
  </div>
</DocumentFragment>
`;
